### PR TITLE
[MIG] web_easy_switch_company: migrate to 9.0

### PR DESCRIPTION
**Pull request by njeudy**
- Use new UserMenu include method
- Works with Community and enterprise

### CODE REVIEW

**Comment by njeudy** [comment]
@rvalyi @pedrobaeza @lmignon  this one can not be used on entreprise too ?

**Comment by pedrobaeza** [comment]
Yeah, it's the same issue. Any module depending on web module will have the same problem. Anyway, have you tried if this module appears good with the enterprise UI?

**Comment by njeudy** [comment]
@pedrobaeza yes it work on community and entreprise. I can post screenshot of the result if needed.

**Comment by njeudy** [comment]
@pedrobaeza need I squash on one commit ?

**Comment by pedrobaeza** [comment]
Well, it's not mandatory in this case, but you should know that the README.md task update is done automatically via an script that checks every module and its installability.

**Comment by legalsylvain** [comment]
Hi @njeudy. Thanks for porting this module.
- I just tested on runbot, and it doesn't seem to work.
  (http://3135224-260-e31796.runbot2.odoo-community.org/) Do you know why ?
- Could you create one demo company, and add admin user to the right groups (demo data) (to make this module more easy to test) ?

Kind regards.

**Comment by njeudy** [comment]
@legalsylvain: as I know you can only test on runbot when it will be merged ?

I look to you runbot instance and it is not le module I propose in this merge request.
- To test, juste add crm app and enable multi-company settings. I can just add a small description on how to test ?

**Comment by njeudy** [comment]
@pedrobaeza ok did know that README.md is autoamticcaly updated :)

**Comment by legalsylvain** [comment]
- @njeudy :
  no, runbot use PR code. you can see your changes [here](http://3135224-260-e31796.runbot2.odoo-community.org/web?db=3135224-260-e31796-all#id=117&view_type=form&model=ir.module.module&menu_id=51&action=36).
- It's better (*) if the module can work just installing the module with demo data. It's interesting to see all the groups needed to make the module works. (but it's "nice to have" and can be done in another PR).
  (better for promotion, test, ...)

**Comment by njeudy** [comment]
Ok, 

To test user should have at least two allowed company in his user form. I think I can add small docs ?

I update administrator account on the runbot instance you indicate ( @legalsylvain )
And it works.

**Comment by njeudy** [comment]
screenshot

**Comment by legalsylvain** [comment]
:+1: Thanks. (code review & test).

**Comment by njeudy** [comment]
need to improve something on this module or is it ok ?

File: README.md [hunk 1]
```diff
@@ -30,7 +30,7 @@ addon | version | summary
 [web_dashboard_tile](web_dashboard_tile/) | 8.0.1.0.0 (unported) | Add Tiles to Dashboard
 [web_dialog_size](web_dialog_size/) | 8.0.0.1.0 (unported) | A module that lets the user expand a dialog box to the full screen width.
 [web_dom_model_classes](web_dom_model_classes/) | 8.0.1.0.0 (unported) | Allows small UI changes with simple CSS
-[web_easy_switch_company](web_easy_switch_company/) | 8.0.1.0.0 (unported) | Multicompany - Easy Switch Company
+[web_easy_switch_company](web_easy_switch_company/) | 9.0.1.0.0 | Multicompany - Easy Switch Company
```
Comment (pedrobaeza) [inline README.md]: This is not in the correct section, so better to leave it without touching, and let the script to update this README.

File: web_easy_switch_company/__openerp__.py [hunk 2]
```diff
@@ -22,7 +22,7 @@
 
 {
     'name': 'Multicompany - Easy Switch Company',
-    'version': '8.0.1.0.0',
+    'version': '9.0.1.0.0',
     'category': 'web',
     'description': """
 Add menu to allow user to switch to another company more easily
```
Comment (pedrobaeza) [inline web_easy_switch_company/__openerp__.py]: Move this description to a README.rst file and use the OCA's latest template: https://raw.githubusercontent.com/OCA/maintainer-tools/master/template/module/README.rst

**Comment by pedrobaeza** [comment]
Please add a disclaimer in the README.rst file indicating that this module is not compatible with web enterprise addon.

**Comment by njeudy** [comment]
@pedrobaeza thanks for your review ... all should be fixed now :)

File: web_easy_switch_company/README.rst [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+   :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+   :alt: License: AGPL-3
+
+=======================
+web_easy_switch_company
+=======================
+
+This module extend web backend and allow user to switch to another company more easyly.
+
+Note: Due to licence incompatibility, you have not the legal right to install this module on enterprise version ...
```
Comment (sbidoul) [inline web_easy_switch_company/README.rst]: According to https://github.com/odoo/odoo/issues/9973#issuecomment-163199596 this sentence is not necessary.

**Comment by njeudy** [comment]
@sbidoul so ? I remove it ? what is OCA advise on this ?

File: web_easy_switch_company/README.rst (hunk 3 above)
Comment (eLBati) [inline web_easy_switch_company/README.rst]: I agree

**Comment by sbidoul** [comment]
@njeudy I _personally_ think you can remove it.

This topic is on the agenda of the next OCA board meeting on January 12th so we should have an official OCA position shortly after.

**Comment by pedrobaeza** [comment]
@njeudy sorry for the noise. It seems that finally it's correct that this module can be used by both community and enterprise (at least this seems so according https://github.com/odoo/odoo/issues/9973#issuecomment-163199596), so no need for the disclaimer.

**Comment by njeudy** [comment]
@pedrobaeza @sbidoul : Ok I rebase on top of V9 and update README.rst without legal claim.

Think it's pretty good now.

File: web_easy_switch_company/__openerp__.py (hunk 2 above)
Comment (pedrobaeza) [inline web_easy_switch_company/__openerp__.py]: Now that you have the README, please remove this key

**Comment by llacroix** [comment]
Any idea why this module isn't yet merged.. ?

File: web_easy_switch_company/static/src/js/switch_company.js [hunk 4]
```diff
@@ -17,18 +18,27 @@
...
-openerp.web_easy_switch_company = function (instance) {
+odoo.define('web.web_easy_switch_company',function (require) {
+
+    "use strict";
+
+    var core = require('web.core');
+    var Widget = require('web.Widget');
+    var UserMenu = require('web.UserMenu');
+    var SystrayMenu = require('web.SystrayMenu');
+    var web_client = require('web.web_client');
```
Comment (llacroix) [inline web_easy_switch_company/static/src/js/switch_company.js:27]: This line isn't necessary and the UserMenu probably too isn't necessary as you're using SystrayMenu

**Comment by njeudy** [comment]
@llacroix thanks ! update done

File: web_easy_switch_company/README.rst [hunk 5]
```diff
@@ -0,0 +1,88 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+   :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+   :alt: License: AGPL-3
+
+=======================
+web_easy_switch_company
+=======================
+
+This module extend web backend and allow user to switch to another company more easyly.
```
Comment (max3903) [inline web_easy_switch_company/README.rst]: This module extends web backend and allows user to switch to another company more easily.

File: web_easy_switch_company/README.rst [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+
+Configuration
+=============
+
+To configure this module, you need to:
+
+* Enable multi company in your odoo instance
+* Add your user in at least 2 company, and you will see the switch at the upper right corner
```
Comment (max3903) [inline web_easy_switch_company/README.rst]: 2 companies

File: web_easy_switch_company/README.rst [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+
+Usage
+=====
+
+Functionality:
+--------------
+    * Add a new menu in the top bar to switch to another company more easily;
+    * Remove the old behaviour to switch company;
```
Comment (max3903) [inline web_easy_switch_company/README.rst]: No space before the star

**Comment by max3903** [comment]
Minor typos. Once they are fixed along with @pedrobaeza's comments, :+1:

**Comment by njeudy** [comment]
Ok I rebase on top of last 9.0 oca/web revision and update to correct all comment.

**Comment by njeudy** [comment]
@pedrobaeza @max3903 @llacroix any chance to merge it ?

:+1:

**Comment by max3903** [comment]
:+1:

**Comment by pedrobaeza** [comment]
:+1:

**Pull request merged**